apm-agent-core/src/main/java/co/elastic/apm/agent/report/ApmServerClient.java

```diff
@@ -121,6 +122,11 @@ private void setServerUrls(List<URL> serverUrls) {
         return copy;
     }
 
+    private static void tlsFallback(HttpsURLConnection connection) {
+        SSLSocketFactory newFactory = SslUtils.getTLSFallbackSocketFactory(connection.getSSLSocketFactory());
```

Comment:
Why don't we create our own SSLSocketFactory from scratch?

Reply:
You mean, why do we wrap the default `SSLSocketFactory` where we could just have created a new one ?

For me that's just a way to make sure that our implementation does not introduce obvious security bugs or unexpected behaviors here. Also, the default implementation is not public thus we can't inherit from it.

Reply:
I mean similar to this: https://gist.github.com/fkrauthan/ac8624466a4dee4fd02f